src/dlfcn.h

```diff
@@ -25,9 +25,18 @@
 #define DLFCN_H
 
 #ifdef __cplusplus
+#include <csetjmp>
+#define EXEC_CXX_FORK(JMPBUF)        std::setjmp(JMPBUF)
+#define EXIT_CXX_FORK(JMPBUF,STATUS) std::longjmp(JMPBUF,STATUS)
 extern "C" {
+#else
+#include <setjmp.h>
 #endif
 
+/* Defined to make things less confusing when find the calls in code */
+#define EXEC_C_FORK(JMPBUF)        setjmp(JMPBUF)
+#define EXIT_C_FORK(JMPBUF,STATUS) longjmp(JMPBUF,STATUS)
+

```

Comment:
If these are just used in the `.c` file, could it make sense to move it to the beginning of `dlfcn.c` so we avoid introducing non-prefixed macros in the public interface of the library?

Reply:
that's fine, just thought you would want it there, could just wrap it in an `#ifdef BUILD_DLFCN` or something

Reply:
Unless I am missing something, it seems to me that moving them to `dlfcn.c` has no downside and it is more robust? Any `#ifdef BUILD_DLFCN`-kind of logic depends on the build system, and I know that several users of `dlfcn-win32` just use `dlfcn.h` and `dlfcn.c` inside their build systems.